homeassistant/components/climate/device_trigger.py

```diff
@@ -0,0 +1,192 @@
+"""Provides device automations for Climate."""
+from typing import List
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_DOMAIN,
+    CONF_TYPE,
+    CONF_PLATFORM,
+    CONF_DEVICE_ID,
+    CONF_ENTITY_ID,
+    CONF_FOR,
+    CONF_ABOVE,
+    CONF_BELOW,
+)
+from homeassistant.core import HomeAssistant, CALLBACK_TYPE
+from homeassistant.helpers import config_validation as cv, entity_registry
+from homeassistant.helpers.typing import ConfigType
+from homeassistant.components.automation import (
+    state as state_automation,
+    numeric_state as numeric_state_automation,
+    AutomationActionType,
+)
+from homeassistant.components.device_automation import TRIGGER_BASE_SCHEMA
+from . import DOMAIN, const
+
+TRIGGER_TYPES = {
+    "current_temperature_changed",
+    "current_humidity_changed",
+    "hvac_mode_changed",
+}
+
+HVAC_MODE_TRIGGER_SCHEMA = TRIGGER_BASE_SCHEMA.extend(
+    {
+        vol.Required(CONF_ENTITY_ID): cv.entity_id,
+        vol.Required(CONF_TYPE): "hvac_mode_changed",
+        vol.Required(state_automation.CONF_TO): vol.In(const.HVAC_MODES),
```

Comment:
@emontnemery what do you suggest we do here. Shall I keep it like this, or should I create a trigger per mode?

Reply:
Discussed on Discord. This is fine.